bin/concurrency/summary_to_requests_per_second.rb

```diff
@@ -16,7 +16,7 @@
 
 summary = JSON.parse(STDIN.read)
 
-requests = summary["requests"].select { |r| r["status"] == 204 }.map do |r|
```

Comment:
This was to limit it to only successful requests which, I think?, is what we care about. I could see plotting the other response codes as separate lines (per the above TODO).

Reply:
Hmm, yeah, I guess we want to see both numbers. The number of successful requests doesn't show if Vector is spamming the destination with extraneous requests. For these tests, I think the number of attempted requests is more important, as it shows Vector's behavior.

Reply:
Yeah, that is a good point, that is an important number.

Maybe we could adjust this to plot the each status code separately.

Reply:
I think the best we could do is to graph requests sent, 200 codes, 400 codes, and 500 codes as four fixed columns/lines. I wonder if that wouldn't become too confusing in the graph though. I'll work on outputting both sent and received, though.

Reply:
Does 4e7ea49 address this for you?

Reply:
👍 I like the changes there (left a couple of comments). However, it does seem to be hanging when plotting for me now, going on two minutes.

Reply:
It is an extremely inefficient algorithm, doing a full pass over the array of timestamps once for each output second. However, I would not expect it to take more than a few seconds each run. I don't see any reason for it to hang, does it show as running on `top`?

Reply:
It did eventually finish, but I have no idea how long it took 😅 . I'll rerun with `time`. I was running:

```
./bin/run-concurrency-test bin/concurrency/suite/429_limit.sh
```

I like the added issued / accepted lines (though I would call "accepted", "successful response", to be more accurate.

However, for the 429 case at least, Vector ends up sending way more than are accepted so it makes it hard to see the successful ones:

![plot](https://user-images.githubusercontent.com/316880/127702890-addfd304-bc73-4b81-be5a-b2b29804a52a.png)

I'm wondering if gnuplot can do separate y-axises to plot them at different scales? Or maybe they should just be separate plots?

Reply:
Right, I have another patch coming to scale that third graph to a reasonable expected range. I'll push that up once I run all the tests to see a good value for the expected rate.

Reply:
Nice, that makes sense.

Regarding plotting time, it seems to take roughly 45 minutes for me locally 😨 

```
time ./bin/run-concurrency-test bin/concurrency/suite/429_limit.sh
writing output files to /var/folders/s5/l4cmrm896l39_65qdvg8yjz00000gn/T/vector-iO7YTiZbWe
starting test server
running test command
./bin/run-concurrency-test: line 56: 22258 Killed: 9               timeout --signal KILL ${TEST_TIME} bash -c "${TEST_CMD}" 2> "${OUTPUT_DIR}/test_cmd.err" > "${OUTPUT_DIR}/test_cmd.log"
processing output
plotting
Fontconfig warning: ignoring UTF-8: not a valid region tag
wrote files to /var/folders/s5/l4cmrm896l39_65qdvg8yjz00000gn/T/vector-iO7YTiZbWe
./bin/run-concurrency-test bin/concurrency/suite/429_limit.sh  2721.73s user 1056.70s system 98% cpu 1:03:45.36 total
```

Something seems no quite right in my environment.

```
➜  vector git:(c5e47320c) ✗ ruby --version
ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
➜  vector git:(c5e47320c) ✗ gnuplot --version
gnuplot 4.6 patchlevel 2
```

I can poke at it a bit more if it doesn't happen for you.

Reply:
Nope, it takes about a second for me. My software is only slightly newer: ruby 2.6.8 and gnuplot 5.4.1